ocaml/xapi/xapi_host.ml

```diff
@@ -1171,6 +1171,24 @@ let install_server_certificate ~__context ~host ~certificate ~private_key ~certi
   (* sever all connections done with stunnel *)
   Xapi_mgmt_iface.reconfigure_stunnel ~__context
 
+let emergency_server_certificate_reset ~__context =
+  let generate_ssl_cert = "/opt/xensource/libexec/generate_ssl_cert" in
+
+  let args = [!Xapi_globs.xapissl_path; Helper_hostname.get_hostname ()] in
+  ignore @@ Forkhelpers.execute_command_get_output generate_ssl_cert args;
+
+  let self = Helpers.get_localhost ~__context in
+
+  (* Delete records of the server certificate in this host *)
+  Db.Certificate.get_all_records ~__context
+  |> List.filter_map (fun (reference, record) ->
```

Comment:
For this kind of operations, take a look at `get_refs_where`. It's a little awkward, but more efficient, since the filtering happens on the master. E.g.

    let expr = Db_filter_types.(Eq (Field "host", Literal (Ref.string_of self))) in
    Db.Certificate.get_refs_where ~__context ~expr